### PR TITLE
Add back __array_wrap__ for dask compatibility

### PR DESCRIPTION
**Pull request by jorisvandenbossche**
xref https://github.com/pandas-dev/pandas/pull/44006/files#r786627658

### CODE REVIEW

File: pandas/tests/base/test_misc.py [hunk 1]
```diff
@@ -84,6 +84,16 @@ def test_ndarray_compat_properties(index_or_series_obj):
     assert Series([1]).item() == 1
 
 
+def test_array_wrap_compat():
+    # Note: at time of dask 2022.01.0, this is still used by eg dask
+    # (https://github.com/dask/dask/issues/8580).
+    # This test is a small dummy ensuring coverage
+    orig = Series([1, 2, 3], index=["a", "b", "c"])
+    result = orig.__array_wrap__(np.array([2, 4, 6]))
```
Comment (jbrockmendel) [inline pandas/tests/base/test_misc.py:92]: is there an actual use case where this gets called organically?

Comment (jorisvandenbossche) [reply pandas/tests/base/test_misc.py:92]: I suppose only with old numpy? (not really an idea though) But so in dask's case they are calling it explicitly / manually (like this test does)

Comment (jbrockmendel) [reply pandas/tests/base/test_misc.py:92]: then wouldn't the reasonable thing to do be either a) see if they really need to call it or b) add a "real" test in test_downstream?

Comment (jorisvandenbossche) [reply pandas/tests/base/test_misc.py:92]: > see if they really need to call it

They don't need to, and are already fixing it (but we should still do this PR to not directly break released dask IMO, and I can open an issue to remove it in a year or so, or we can first deprecate it)

> add a "real" test in test_downstream?

I would say that the "real" test is downstream in dask's test suite :) 
In the end I don't really care where the test is put, but I think putting it here without usage of dask is useful (I think the dowstream tests are much easier to miss while developing)

Comment (jorisvandenbossche) [reply pandas/tests/base/test_misc.py:92]: Added a downstream dask test, but also kept this one.

Will do a follow-up PR to add a deprecation warning for 1.5

**Comment by jreback** [comment]
lots of things failing here, can you merge master

File: pandas/tests/test_downstream.py [hunk 2]
```diff
@@ -55,6 +55,30 @@ def test_dask(df):
         pd.set_option("compute.use_numexpr", olduse)
 
 
+@pytest.mark.filterwarnings("ignore:.*64Index is deprecated:FutureWarning")
```
Comment (jreback) [inline pandas/tests/test_downstream.py:58]: can you use the import decorator here e.g. `@td.skip_if_no("xarray"` (and can you fix the L39 one

i also don't know why we have a special import_module here.

Comment (jorisvandenbossche) [reply pandas/tests/test_downstream.py:58]: This seems to be done on purpose:

https://github.com/pandas-dev/pandas/blob/4ce84de5250aafa706d4c7610fdefad0baf37acf/pandas/tests/test_downstream.py#L23-L30

Comment (jreback) [reply pandas/tests/test_downstream.py:58]: ok really this should be cleaned up cc @jbrockmendel

**Comment by jreback** [comment]
@meeseeksdev backport 1.4.x

File: pandas/tests/base/test_misc.py (hunk 1 above)
Comment (jorisvandenbossche) [reply pandas/tests/base/test_misc.py]: Follow-up PR to deprecate it -> https://github.com/pandas-dev/pandas/pull/45517

**Pull request merged**